consul/client.go

```diff
@@ -215,7 +229,13 @@ func (c *Client) Encrypted() bool {
 
 // lanEventHandler is used to handle events from the lan Serf cluster
 func (c *Client) lanEventHandler() {
+	var numQueuedEvents int
 	for {
+		numQueuedEvents = len(c.eventCh)
+		if numQueuedEvents > serfEventBacklogWarning {
```

Comment:
Since you measure the length here anyway I wonder if it's better to emit a stat vs. having the warning mechanism. I wish we could set "alert" metadata on stats :-)